### DISCUSSION TITLE
Infinite scroll test

### DISCUSSION THREAD

**Discussion by baeharam** [cypress-io/cypress#15370] [Questions and Help] [answered]
I have infinite list and when I scroll to bottom, it makes request to server to get a new items.
So I wanna test like this,

1) Manually create dummy items to real server
2) Dummy items could be mixed and not sorted well, so I have to find item that I made for test manually. It means that I have to scroll to bottom until item is found.

In this situation, how can I write test properly? Is it best practice? I wanna test like below.

```
while(cy.contains(newItem).should('be.visible')) {
  cy.scrollTo('bottom'); // scroll and server request
}
```

I know this is wrong for Cypress syntax, but I just wrote for explanation. Thanks.

**Comment by bahmutov** [comment, marked as answer]
Please look at the recipe https://github.com/cypress-io/cypress-example-recipes/tree/master/examples/testing-dom__page-reloads - you never want to use the `while` loop, since this does not work with asynchronous Cypress commands.
